scripts/oneoff/update-supplier-data.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+"""Usage: update-supplier-data.py <path-to-updates> [--stage=STAGE] [--attribute=ATTRIBUTE] [--dry-run]
+
+This script iterates through a CSV and updates a supplier with one of three values: DUNS number, company registration
+number, or company name.
+
+The DUNS number is an interesting problem, as there are cases where suppliers have used a number that belongs to another
+supplier. When the rightful owner attempts to add it, they are told they can't as it's already in use. This is due to a
+UNIQUE restriction on the database column. For now, those cases will simply be logged and you'll need to work out what
+to do with them separately.
+
+Options:
+    <path-to-updates>                                     The path to the CSV containing updates
+
+    [--stage=STAGE]                                       Stage to target
+
+    [--attribute=ATTRIBUTE]                               Attribute to update
+
+    [--dry-run]                                           List actions that would have been taken
+
+    -h, --help                                            Show this screen
+
+"""
+import sys
+import csv
+import re
+from dmapiclient.data import DataAPIClient
+
+sys.path.insert(0, '.')
+
+from dmscripts.helpers.auth_helpers import get_auth_token
+from dmutils.env_helpers import get_api_endpoint_from_stage
+from docopt import docopt
+from pathlib import Path
+from dmapiclient.errors import HTTPError
+
+
+UPDATER = "update supplier data script"
+
+
+def update_supplier_attribute(row):
+    attribute_map = {
+        "DUNS Number": update_duns,
+        "Company registration number": update_company_number,
+        "Company registered name": update_company_name,
+    }
+    supplier_id = row.get("DMP record ID")
+    new_value = row.get("New Value")
+    field_to_change = row.get("Field Name for update")
+    try:
+        attribute_map.get(field_to_change)(supplier_id, new_value)
+    except HTTPError as e:
+        print(f"Error updating supplier {supplier_id} with error {e}")
+
+
+def update_duns(supplier_id, new_duns):
+    try:
+        data_api_client.update_supplier(supplier_id, {"dunsNumber": new_duns}, UPDATER)
```

Comment:
Can we have some log output on the happy path too - so we have an idea of what's happening when we run for real?